### PR TITLE
Hide non-specd attributes for aws-sdk-2.2

### LINKED ISSUE #1874: Hide all span attributes not defined by spec behind an experimental flag
Otherwise, if/when they are defined by the spec, we will have to keep the old ones around forever too if we don't want to introduce a breaking change.

### PR DESCRIPTION
**Pull request by trask**
Closes #1874

### CODE REVIEW

File: instrumentation/external-annotations/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/extannotations/TraceAnnotationsInstrumentationModule.java [hunk 1]
```diff
@@ -125,20 +126,19 @@ public AnnotatedMethodsInstrumentation() {
           TraceAdvice.class.getName());
     }
 
-    private static Set<String> configureAdditionalTraceAnnotations(Config config) {
-      String configString = config.getProperty(TRACE_ANNOTATIONS_CONFIG);
-      if (configString == null) {
+    private static Set<String> configureAdditionalTraceAnnotations(String include) {
```
Comment (iNikem) [inline instrumentation/external-annotations/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/extannotations/TraceAnnotationsInstrumentationModule.java:129]:
```suggestion
    private static Set<String> configureAdditionalTraceAnnotations(String config) {
```

**Review by iNikem — APPROVED** [review]

**Comment by trask** [comment]
Moving this back to draft status until we decide on #2444

**Comment by anuraaga** [comment]
/cc @mateuszrzeszutek 

For this and #2467 I'm hoping they're temporary, and it's ok. I'd expect us to make these programmatic eventually though. After #2463 I'm planning on tackling aws-sdk-2.2 next, and have an `AwsSdkTracingBuilder.setEnableExperimentalAttributes` type of pattern in mind.

**Comment by mateuszrzeszutek** [comment]
> I'm planning on tackling aws-sdk-2.2 next, and have an `AwsSdkTracingBuilder.setEnableExperimentalAttributes` type of pattern in mind.

That sounds great - and if we refactor all library instrumentations to use this pattern we might be able to remove `Config` from instrumentation-api

**Review by anuraaga — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**